test/events.js

```diff
@@ -1,92 +1,79 @@
-const getCenterPoint = (xScale, yScale, element) => element.getCenterPoint();
-
-export function testEvents(options, eventIn, eventOut) {
-  testEnterEvent(options, 1, eventIn);
-  testLeaveEvent(options, 1, eventOut);
-  testClickEvent(options, 1, eventIn);
-  testEnterEvent(options, 0, eventOut);
-  testLeaveEvent(options, 0, eventIn);
-  testClickEvent(options, 0, eventOut);
-}
-
-function testEnterEvent(options, toBe, getEventPoint) {
-  const context = getTestCaseContext(toBe);
+export const getCenterPoint = (chart) => window.getAnnotationElements(chart)[0].getCenterPoint();
+export const eventPoint0 = {
+  x: 0,
+  y: 0
+};
 
+export function testEvents(options) {
   describe('events', function() {
-    const pluginOpts = context.chartConfig.options.plugins.annotation;
-
-    [pluginOpts, options].forEach(function(targetOptions) {
-
-      it(`${context.description} detect enter event`, function(done) {
-        const enterSpy = jasmine.createSpy('enter');
-
-        targetOptions.enter = enterSpy;
-        pluginOpts.annotations = [options];
-
-        const chart = window.acquireChart(context.chartConfig);
-        const eventPoint = retrieveEventPoint(chart, getEventPoint);
-
-        window.triggerMouseEvent(chart, 'mousemove', eventPoint);
-        window.afterEvent(chart, 'mousemove', function() {
-          expect(enterSpy.calls.count()).toBe(context.compare);
-          delete targetOptions.enter;
-          done();
-        });
-      });
-    });
-  });
-}
-
-function testLeaveEvent(options, toBe, getEventPoint) {
-  const context = getTestCaseContext(toBe);
+    const chartConfig = {
+      type: 'scatter',
+      options: {
+        animation: false,
+        scales: {
+          x: {
+            display: false,
+            min: 0,
+            max: 10
+          },
+          y: {
+            display: false,
+            min: 0,
+            max: 10
+          }
+        },
+        plugins: {
+          legend: false,
+          annotation: {
+          }
+        }
+      },
+    };
 
-  describe('events', function() {
-    const pluginOpts = context.chartConfig.options.plugins.annotation;
+    const pluginOpts = chartConfig.options.plugins.annotation;
 
     [pluginOpts, options].forEach(function(targetOptions) {
 
-      it(`${context.description} detect leave event`, function(done) {
+      it(`should detect enter and leave events on ${options.type}`, function(done) {
         const enterSpy = jasmine.createSpy('enter');
         const leaveSpy = jasmine.createSpy('leave');
 
         targetOptions.enter = enterSpy;
         targetOptions.leave = leaveSpy;
         pluginOpts.annotations = [options];
 
-        const chart = window.acquireChart(context.chartConfig);
-        const eventPoint = retrieveEventPoint(chart, getCenterPoint);
+        const chart = window.acquireChart(chartConfig);
```

Comment:
could use the `scatter10x10` helper

Reply:
> could use the `scatter10x10` helper

`scatter10x10` is returning a chart already acquired but we need to manage the plugin configuration (not only the annotation one) because we are testing the events hooks at `annotation` node level.

Reply:
I forgot that `scatter10x10` is accepting only annotations as argument.